Makefile

```diff
@@ -143,7 +143,16 @@ prow-ci-manifests-check: manifests
 		exit 1; \
 	fi
 
+GO_MOD_DIRS=. dpu-api api tools
+
+.PHONY: tidy
+tidy:
+	for d in $(GO_MOD_DIRS) ; do \
+		(cd $$d && go mod tidy) || exit $$? ; \
+	done
+
 .PHONY: generate
+generate: tidy
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
can we have these on the same line? Also, can we make it depend on vendor and remove tidy target?

Reply:
ok...

there is a minor problem that running `make vendor` before the `make generate` step, is that that can result in something, which still gets changed by another `make vendor`. But that is only in the case where `make generate` even does anything. So, in some cases, you may still want to run `make generate && make vendor`, although that really runs `make vendor && make generate && make vendor`. But I think that's OK (and not the common case).